docs/rollup-finality-provider-operation.md

```diff
@@ -0,0 +1,798 @@
+}
+```
+
+### 4.3. Configure Your Finality Provider
+
+Edit the `fpd.conf` file in your finality provider home directory with the
+following parameters:
+
+```shell
+[Application Options]
+RollupNodeRPCAddress = <rollup-bsn-rpc-endpoint>
+FinalityContractAddress = <rollup-bsn-contract-address>
+EOTSManagerAddress = 127.0.0.1:12582
+RPCListener = 127.0.0.1:12581
+
+[babylon]
+Key = <finality-provider-key-name-signer> # the key you used above
+ChainID = bbn-test-5 # chain ID of the Babylon Genesis
+RPCAddr = http://127.0.0.1:26657 # Your Babylon Genesis node's RPC endpoint
+KeyDirectory = <path> # The `--home` path to the directory where the keyring is stored
+```
+
+> ⚠️ **Important**: Operating a finality provider requires direct 
+> connections to both a Babylon Genesis full node and a Rollup BSN node. 
+> It is **highly recommended** to operate your own instances of
+> full nodes instead of relying on third parties. You can find
+> instructions on setting up a Babylon Genesis node
+> [here](https://github.com/babylonlabs-io/networks/tree/main/bbn-test-5/babylon-node/README.md).
+>
+> ⚠️ **Critical RPC Configuration**:
+> When configuring your finality provider to a Babylon Genesis RPC node, you should
+> connect to a **single** node directly. Additionally you **must**
+> ensure that this node has transaction indexing enabled (`indexer = "kv"`).
+> Using multiple RPC nodes or load balancers can lead to sync issues.
+
+> ⚠️ **Contract Requirement**: To start a finality provider you must 
+> supply a valid Rollup BSN contract address. Additionally, the `btc_pk` obtained 
+> from `eotsd` must be included in the contract’s allow-list. If the contract 
+> address is missing/invalid or the `btc_pk` isn’t allowed, the finality provider 
+> will fail to start.
+
+Configuration parameters explained:
+
+* `RollupNodeRPCAddress`: Your Rollup BSN node's RPC endpoint
+* `FinalityContractAddress`: Address of the rollup finality contract
+* `EOTSManagerAddress`: Address where your EOTS daemon is running
+* `RPCListener`: Address for the finality provider RPC server
+* `Key`: Your Babylon Genesis key name from Step 2
+* `ChainID`: The Babylon Genesis chain ID
+* `RPCAddr`: Your Babylon Genesis node's RPC endpoint
+* `KeyDirectory`: Path to your keyring directory (same as `--home` path)
+
+Please verify the `chain-id` and other network parameters from the official
+[Babylon Genesis Networks
+repository](https://github.com/babylonlabs-io/networks).
+
+Another notable configurable parameter is `NumPubRand`, which is the number of
+public randomness that will be generated and submitted in one commit to Babylon
+Genesis. This value is set to `50,000` by default, which is sufficient for
+roughly 5 days of usage with block production time at `10s`.
+Larger values can be set to tolerate longer downtime with larger size of
+merkle proofs for each randomness, resulting in higher gas fees when submitting
+future finality signatures and larger storage requirements.
```

Comment:
I think this paragraph needs to be double checked depending on the rollup BSN contract parameters.

Reply:
It all depends how in detail we want to explain. Rollup BSN contract has submissionInterval. And depending on this interval fp will submit signature every n-th block. If interval is set to 1, then this is correct and is the same as with babylon FP. But if interval is set to 5, then number of days is multiplied by 5.

Reply:
I mean whether this description still makes sense for rollup bsns